okhttp-testing-support/src/main/kotlin/okhttp3/RecordingEventListener.kt

```diff
@@ -69,8 +69,12 @@ open class RecordingEventListener(
    */
   private val enforceOrder: Boolean = true,
 ) : EventListener() {
+  /** Events that haven't yet been removed. */
   val eventSequence: Deque<CallEvent> = ConcurrentLinkedDeque()
 
+  /** The full set of events, used to match starts with ends. */
+  private val eventsForMatching = ConcurrentLinkedDeque<CallEvent>()
```

Comment:
(My test case consumes events before the test completes, and that was breaking matching.)